src/main/java/net/earthcomputer/clientcommands/command/ItemGroupCommand.java

```diff
@@ -0,0 +1,307 @@
+import net.fabricmc.fabric.api.util.NbtType;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.SharedConstants;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.datafixer.TypeReferences;
+import net.minecraft.item.ItemGroup;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.*;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.Util;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mojang.brigadier.arguments.IntegerArgumentType.*;
+import static com.mojang.brigadier.arguments.StringArgumentType.*;
+import static net.earthcomputer.clientcommands.command.ClientCommandManager.*;
+import static net.minecraft.command.CommandSource.*;
+import static net.minecraft.command.argument.ItemStackArgumentType.*;
+import static net.minecraft.server.command.CommandManager.*;
+
+public class ItemGroupCommand {
+
+    private static final Logger LOGGER = LogManager.getLogger("clientcommands");
+
+    private static final DynamicCommandExceptionType NOT_FOUND_EXCEPTION = new DynamicCommandExceptionType(arg -> new TranslatableText("commands.citemgroup.notFound", arg));
+    private static final DynamicCommandExceptionType OUT_OF_BOUNDS_EXCEPTION = new DynamicCommandExceptionType(arg -> new TranslatableText("commands.citemgroup.outOfBounds", arg));
+
+    private static final SimpleCommandExceptionType SAVE_FAILED_EXCEPTION = new SimpleCommandExceptionType(new TranslatableText("commands.citemgroup.saveFile.failed"));
+    private static final DynamicCommandExceptionType ILLEGAL_CHARACTER_EXCEPTION = new DynamicCommandExceptionType(arg -> new TranslatableText("commands.citemgroup.addGroup.illegalCharacter", arg));
+    private static final DynamicCommandExceptionType ALREADY_EXISTS_EXCEPTION = new DynamicCommandExceptionType(arg -> new TranslatableText("commands.citemgroup.addGroup.alreadyExists", arg));
+
+    private static final Path configPath = FabricLoader.getInstance().getConfigDir().resolve("clientcommands");
+
+    private static final MinecraftClient client = MinecraftClient.getInstance();
+
+    private static final Map<String, Group> groups = new HashMap<>();
+
+    static {
+        try {
+            loadFile();
+        } catch (IOException e) {
+            LOGGER.info("Could not load groups file, hence /citemgroup will not work!");
+        }
+    }
+
+    public static void register(CommandDispatcher<ServerCommandSource> dispatcher) {
+        addClientSideCommand("citemgroup");
+
+        LiteralCommandNode<ServerCommandSource> citemgroup = dispatcher.register(literal("citemgroup"));
+        dispatcher.register(literal("citemgroup")
```

Comment:
Can some of these indents below are twice as much as they should be, code shouldn't the this far right!